plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -512,9 +512,9 @@ public PartitionStatistics getTableStatistics(Table table)
         PartitionStatistics missingColumnStatistics = delegate.getTableStatistics(tableWithOnlyMissingColumns);
 
         // merge missing stats into the cached stats
-        return tableStatisticsCache.asMap().compute(
-                tableName,
-                (ignored, oldStats) -> {
+        AtomicReference<PartitionStatistics> currentReference = uncheckedCacheGet(tableStatisticsCache, tableName, AtomicReference::new);
```

Comment:
There is still a race condition between invalidate and `delegate.getTableStatistics(tableWithOnlyMissingColumns);`.

It might happen that after `delegate.getTableStatistics(tableWithOnlyMissingColumns);` there is stats invalidation.
Then `uncheckedCacheGet(tableStatisticsCache, tableName, AtomicReference::new);` would create new atomic reference, but put old value into cache.

Reply:
moved to the new `getWithValueHolder` and fixed